File: backend/lambdas/broadcast-lambda/src/main/java/com/artcorner/erp/dto/response/orders/OrderResponseForEmployee.java
package com.artcorner.erp.dto.response.orders;

import com.artcorner.erp.entities.orders.OrderStatus;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class OrderResponseForEmployee {
    private Long orderId;
    private OrderStatus orderStatus;
    private BigDecimal totalAmount;

    private UUID customerId;
    private String customerName;
    private String customerEmail;
    private String customerPhone;
    private List<OrderItemResponse> orderItemResponses;

    @Data
    @Builder
    public static class OrderItemResponse {
        private Long itemId;
        private Long productId;
        private String productName;
        private int quantity;
        private BigDecimal price;
    }
}

File: frontend/Employees/src/features/orders/types/OrderResponseForEmployee.ts
import { OrderStatus } from "./OrderStatus";

export interface OrderResponseForEmployee {
  orderId: number;
  orderStatus: OrderStatus;
  totalAmount: number;

  customerId: string;
  customerName: string;
  customerEmail: string;
  customerPhone: string;

  orderItemResponses: OrderItemResponse[];
}

export interface OrderItemResponse {
  itemId: number;
  productId: number;
  productName: string;
  quantity: number;
  price: number;
}
